bootloader/bootloader.go

```diff
@@ -343,3 +343,40 @@ func ForGadget(gadgetDir, rootDir string, opts *Options) (Bootloader, error) {
 	}
 	return nil, ErrBootloader
 }
+
+// BootImage represents each image in the chains of trusted assets and kernels
+// used in the boot process. A boot image can be an EFI binary or a snap file
+// containing an EFI binary.
+type BootImage struct {
+	// Path is the path to the boot image file.
+	Path string
+	// Relative is the path to the EFI image inside a snap container.
+	Relative string
+	// FromRecovery is true if this boot image originates from the recovery
+	// bootloader boot chain.
+	FromRecovery bool
```

Comment:
this should become a Role from Role PR I think

Reply:
Will update to Role when it becomes available.